lib/widgets/payment_request_dialog.dart

```diff
@@ -10,58 +9,183 @@ import 'package:flutter_advanced_networkimage/flutter_advanced_networkimage.dart
 import 'package:image/image.dart' as DartImage;
 import 'package:breez/bloc/account/account_model.dart';
 import 'package:auto_size_text/auto_size_text.dart';
+import 'package:breez/widgets/loading_animated_text.dart';
+
+enum PaymentRequestState { PAYMENT_REQUEST, WAITING_FOR_CONFIRMATION, PROCESSING_PAYMENT}
+final key = new GlobalKey<PaymentRequestDialogState>();
 
 class PaymentRequestDialog extends StatefulWidget {
   final BuildContext context;
   final AccountBloc accountBloc;
   final PaymentRequestModel invoice;
   final _transparentImage = DartImage.encodePng(DartImage.Image(300, 300));
 
-  PaymentRequestDialog(this.context, this.accountBloc, this.invoice);
+  PaymentRequestDialog(this.context, this.accountBloc, this.invoice, Key key)
+      : super(key: key);
 
   @override
-  State<StatefulWidget> createState() {    
+  State<StatefulWidget> createState() {
     return PaymentRequestDialogState();
   }
 }
 
-class PaymentRequestDialogState extends State<PaymentRequestDialog> {
+class PaymentRequestDialogState extends State<PaymentRequestDialog>
+    with SingleTickerProviderStateMixin {
   final _formKey = GlobalKey<FormState>();
   TextEditingController _invoiceAmountController = new TextEditingController();
 
+  AnimationController controller;
+  Animation<double> opacityAnimation;
+  Animation<double> borderAnimation;
+  Animation<RelativeRect> transitionAnimation;
+  Animation<Color> colorAnimation;
+
+  PaymentRequestState _state;
+
+  Int64 _amountToPay;
+  String _amountToPayStr;
+
+  bool _isInit = false;
+
   @override
-  void initState() {    
+  void initState() {
     super.initState();
-    _invoiceAmountController.addListener((){
+    setState(() {
+      _state = PaymentRequestState.PAYMENT_REQUEST;
+    });
+    controller = AnimationController(
+        vsync: this, duration: Duration(milliseconds: 600));
+    borderAnimation = Tween<double>(begin: 0.0, end: 8.0).animate(
+        CurvedAnimation(parent: controller, curve: Curves.fastOutSlowIn));
+    opacityAnimation = Tween<double>(begin: 0.0, end: 1.0).animate(
+        CurvedAnimation(parent: controller, curve: Curves.fastOutSlowIn));
+    colorAnimation = new ColorTween(
+      begin: theme.BreezColors.blue[500],
+      end: theme.BreezColors.white[500],
+    ).animate(controller)
+      ..addListener(() {
+        setState(() {});
+      });
+    _invoiceAmountController.addListener(() {
       setState(() {});
     });
   }
 
+  @override
+  void didChangeDependencies() {
```

Comment:
All this method content can be re-located into initState method.
The only reason to use didChangDependencies is when we need to fetch from the inheritedWidget which is not available in initState.

Reply:
We are using didChangeDependencies to fetch values of inheritedWidget to initialize transitionAnimation, but we can remove the status listener of animation controller to initState.